platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapGestureDetector.java

```diff
@@ -480,7 +480,7 @@ public boolean onMoveBegin(@NonNull MoveGestureDetector detector) {
     @Override
     public boolean onMove(@NonNull MoveGestureDetector detector, float distanceX, float distanceY) {
       // first move event is often delivered with no displacement
-      if (distanceX != 0 || distanceY != 0) {
+      if (!Float.isNaN(distanceX) && !Float.isNaN(distanceY) && (distanceX != 0 || distanceY != 0)) {
```

Comment:
Good find! For short term solution this is a good fix.

This will fix the immediate crash. But there are some questions:

Why is this being called when distances are NAN? Do we call it or is this called externally.

In both cases silently ignoring the call may cause problems later. In some ways I prefer an exception, that will indicator that something is wrong with the code. If we ignore it it may manifest later in a more subtle errors that will be harder to diagnose.

At least we should log the error. because one way or another it is wrong to call this when the distances are not a number.

Reply:
We can probably also check / log the onMoveEnd parameters. I suspect they could also be NAN. But the best thing would be to ensure they are not NAN. unless they come from external caller.

Reply:
So far I haven't been able to reproduce this error, so I am not 100% sure what might have happened here.

My assumption was that this is a similar case to the one mentioned in the comment

`// first move event is often delivered with no displacement`

If you want I can add a logging statement? We can also leave this one open for now, so far there has only been one instance of this error, so it could be that it was some special edge case.

Reply:
I say log an error and merge. We can keep an eye on it if it happens more often.

Reply:
Ok, I added a log :)